test/unit/plugins/validate-semantic/oas3/refs.js

```diff
@@ -206,4 +206,109 @@ describe("validation plugin - semantic - oas3 refs", () => {
       return expectNoErrorsOrWarnings(spec)
     })
   })
-})
+
+  describe("$refs for headers must reference a header by position", () => {
+    it("should return an error when a header incorrectly references a local parameter schema", () => {
+      const spec = {
+        openapi: "3.0.0",
+        paths: {
+          "/foo": {
+            get: {
+              responses: {
+                "200": {
+                  description: "OK",
+                  headers: {
+                    "X-MyHeader": {
+                      $ref: "#/components/parameters/MyHeader"
+                    }
+                  }
+
+                }
+              }
+            }
+          }
+        },
+        components: {
+          headers: {
+            MyHeader: {
+              $ref: "#/components/headers/MyHeader"
+            }
+          }
+        }
+      }
+
+      return validateHelper(spec)
+        .then(system => {
+          const allErrors = system.errSelectors.allErrors().toJS()
+          const firstError = allErrors[1]
+          expect(allErrors.length).toEqual(2)
+          expect(firstError.message).toEqual(`OAS3 response header $refs should point to #/components/headers/... and not #/components/parameters/...`)
+          expect(firstError.path).toEqual(["paths", "/foo", "get","responses","200", "headers", "X-MyHeader", "$ref"])
+        })
+    })
+    
+    it("should return no errors when a header correctly references a local component header", () => {
```

Comment:
Consider changing "header" -> "response header" to avoid confusion with header parameters.
```suggestion
    it("should return no errors when a response header correctly references a local header component", () => {
```

Reply:
Done!